Fix schema and it's validation

buildersbrewery opened the pull request:
This commit ...

- updates the workflow that produces test output so  that the YAML file is checked against the schema first
- uses JSON Schema's format instead of `yamale`'s so we can
  - use LSP with LSP-json to validate the schema
  - and use LSP with LSP-yaml to validate YAML code
- adds an option to validate the YAML file locally against the schema
- closes #33, addresses changes proposed there by including them in the new format used here

buildersbrewery commented:
Anybody have a regex for argument names?

Does something like this work for LSL? 

<img width="762" height="139" alt="Sublime Text 2025-12-03 21 54 58" src="https://github.com/user-attachments/assets/3da0282d-abf5-4b8d-8c44-6bbbb550f66f" />

WolfGangS commented on lsl_definitions.schema.json:
```diff
@@ -0,0 +1,467 @@
+                            "default": false
+                        },
+                        "mono-sleep": {
+                            "$ref": "#definitions/nonNegativeNumber"
```
Missing `/` after `#`
`#/definitions/nonNegativeNumber`

WolfGangS commented on lsl_definitions.schema.json:
```diff
@@ -0,0 +1,467 @@
+                            ]
+                        },
+                        "sleep": {
+                            "$ref": "#definitions/nonNegativeNumber"
```
Missing `/` after `#`
`#/definitions/nonNegativeNumber`

WolfGangS replied:
```suggestion
                            "$ref": "#/definitions/nonNegativeNumber"
```

WolfGangS commented on lsl_definitions.schema.json:
```diff
@@ -0,0 +1,467 @@
+            "type": "object"
+        },
+        "llsd-lsl-syntax-version": {
+            "$ref": "#definitions/positiveInteger"
```
Missing `/` after `#`
`#/definitions/positiveInteger`

buildersbrewery commented on lsl_definitions.schema.json:
```diff
@@ -0,0 +1,467 @@
+                }
+            },
+            "type": "object"
+        },
```
Why are the values of the math constants in string format in this YAML file? Was there a technical reason? Can I just change them to use a float value in the YAML file and change the JSON schema for the YAML to use `number`?

<img width="548" height="268" alt="image" src="https://github.com/user-attachments/assets/521c5c89-642a-483c-8cce-c8061b25e124" />

WolfGangS replied:
I suspect this file was originally generated from ll's llsd XML format. I might be wrong.

But if it was, pretty much everything in XML is a string till you use meta data to cast it. So possible it was just read as a string from the XML.

HaroldCindy commented on lsl_definitions.yaml:
```diff
@@ -2164,15 +2168,15 @@ constants:
   PI_BY_TWO:
     tooltip: 1.57079633 - The number of radians in a quarter circle.
     type: float
-    value: '1.57079633'
```
Odd as it meant seem, floating point values are meant to be strings in this file. The values there are originally extracted from C++ files in the simulator, and we want to make sure the the interpretation of the floating point values doesn't change until the ultimate point-of-use.

HaroldCindy commented on lsl_definitions.yaml:
```diff
@@ -3973,7 +3977,7 @@ constants:
     tooltip: 6.28318530 - The radians of a circle.
     type: float
-    value: '6.28318530'
+    value: 6.28318530
```
see above about floating point

HaroldCindy commented on lsl_definitions.yaml:
```diff
@@ -3653,7 +3657,7 @@ constants:
     tooltip: 1.41421356 - The square root of 2.
     type: float
-    value: '1.41421356'
+    value: 1.41421356
```
see above about floating point

buildersbrewery commented:
I have addressed all comments above. Haven't tested myself locally, yet. But everything is ready to go.

I'll switch to "ready for review" once I've tested locally, but feel free to do it yourself and provide feedback in the meantime.

buildersbrewery commented on pyproject.toml:
```diff
@@ -24,20 +28,29 @@ classifiers = [
+]
+keywords = ["Linden Scripting Language", "Second Life"]
+
+[project.optional-dependencies]
```
moved `jsonschema` to optional dependencies as requested

HaroldCindy commented:
Looks like CI is failing, think the command name in the workflow def needs updating

HaroldCindy commented on validate.py:
```diff
@@ -0,0 +1,34 @@
+    try:
+        jsonschema.validate(instance=yamldata, schema=schemadata)
+        print("\nSUCCESS: Valid against the schema")
+        return True
```
When this is used in utility script form, I believe the return of the function is treated as if it were an integer and used as the return code of the script. Would explain why CI is failing even with a valid schema.

HaroldCindy reviewed (approved):
LGTM, thanks!
